test/tstDetailsHalfTraversal.cpp

```diff
@@ -0,0 +1,106 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "ArborXTest_StdVectorToKokkosView.hpp"
+#include "ArborX_EnableDeviceTypes.hpp" // ARBORX_DEVICE_TYPES
+#include "ArborX_EnableViewComparison.hpp"
+#include <ArborX_DetailsHalfTraversal.hpp>
+#include <ArborX_LinearBVH.hpp>
+
+#include <Kokkos_Random.hpp>
+
+#include "BoostTest_CUDA_clang_workarounds.hpp"
+#include <boost/test/unit_test.hpp>
+
+namespace Test
+{
+template <class ExecutionSpace>
+// enclosing parent function for an extended __host__ __device__ lambda must not
+// have deduced return type
+#ifndef KOKKOS_COMPILER_NVCC
+auto
+#else
+Kokkos::View<ArborX::Point *, ExecutionSpace>
+#endif
+    make_random_cloud(ExecutionSpace const &space, int n)
```

Comment:
We have something similar in `tstQueryTreeComparisonWithBoost.cpp` and `tstDistributedTree.cpp`. Maybe we should just provide a utility. We also have random clouds in `benchmark/point_clouds.hpp`, though may be overkill.

Reply:
Doesn't need an actual random cloud